flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/BarrierBuffer.java

```diff
@@ -122,8 +124,8 @@
 	 *
 	 * @throws IOException Thrown, when the spilling to temp files cannot be initialized.
 	 */
-	public BarrierBuffer(InputGate inputGate, BufferBlocker bufferBlocker) throws IOException {
-		this (inputGate, bufferBlocker, -1);
+	public BarrierBuffer(InputGate inputGate, BufferBlocker bufferBlocker) {
```

Comment:
It looks as if this constructor is only used for testing. Please annotate with `@VisibleForTesting`